src/Main.hs

```diff
@@ -214,6 +217,19 @@ readDerivation path = do
         _ -> do
             fail ("Could not parse a derivation from this file: " ++ string)
 
+-- | Read and parse a derivation from a store path that can be a derivation
+-- (.drv) or a realized path, in which case the corresponding derivation is
+-- queried.
+readInput :: FilePath -> Diff (Derivation FilePath Text)
+readInput path =
+    let string = path in
+    if List.isSuffixOf ".drv" string
+    then readDerivation path
+    else do
+        result <- liftIO (Process.readProcess "nix-store" [ "-qd", string ] [])
+        let drv_path = last (String.lines result)
```

Comment:
`String.lines` can return an empty list if `result` is the empty string, which in turn would cause `last` to fail with an unhelpful error.  So I suggest handling that case by providing an error message.  e.g. something like:

```haskell
case String.lines result of
    [] -> -- error message goes here
    l : ls -> do
        let drv_path = Data.List.NonEmpty.last (l :| ls)
        readDerivation drv_path
```

Reply:
That's really unexpected so the error message can't be more useful but it's still better than failing at `last`.